apps/src/code-studio/components/AbuseExclamation.jsx

```diff
@@ -1,96 +1,40 @@
 import PropTypes from 'prop-types';
 import React from 'react';
 import AbuseError from './AbuseError';
+import AlertExclamation from './AlertExclamation';
+
+export default function AbuseExclamation({i18n, isOwner}) {
+  let finalLink, finalLinkText;
+  if (isOwner) {
+    finalLink = 'edit';
+    finalLinkText = i18n.edit_project;
+  } else {
+    finalLink = 'https://studio.code.org';
+    finalLinkText = i18n.go_to_code_studio;
+  }
 
-/**
- * A big blue box with an exclamation mark on the left and our abuse text on
- * the right.
- */
-export default class AbuseExclamation extends React.Component {
-  static propTypes = {
-    i18n: PropTypes.shape({
-      tos: PropTypes.string.isRequired,
-      contact_us: PropTypes.string.isRequired,
-      edit_project: PropTypes.string.isRequired,
-      go_to_code_studio: PropTypes.string.isRequired
-    }).isRequired,
-    isOwner: PropTypes.bool.isRequired
+  const textStyle = {
+    fontSize: 18,
+    lineHeight: '24px',
+    padding: 5
   };
 
-  render() {
-    const cyan = '#0094ca';
-    const style = {
-      backgroundColor: cyan,
-      color: 'white',
-      maxWidth: 600,
-      margin: '0 auto',
-      marginTop: 20,
-      borderRadius: 15
-    };
-
-    const circleStyle = {
-      width: 100,
-      height: 100,
-      background: 'gold',
-      borderRadius: 50,
-      MozBorderRadius: 50,
-      WebkitBorderRadius: 50,
-      margin: 20,
-      position: 'relative'
-    };
-
-    const exclamationStyle = {
-      fontSize: 80,
-      position: 'absolute',
-      top: '50%',
-      left: '50%',
-      transform: 'translate(-50%, -50%)'
-    };
-
-    const bodyStyle = {
-      paddingLeft: 0,
-      paddingTop: 10,
-      paddingBottom: 10,
-      paddingRight: 20
-    };
-
-    const textStyle = {
-      fontSize: 18,
-      lineHeight: '24px',
-      padding: 5
-    };
-
-    let finalLink, finalLinkText;
-    if (this.props.isOwner) {
-      finalLink = 'edit';
-      finalLinkText = this.props.i18n.edit_project;
-    } else {
-      finalLink = 'https://studio.code.org';
-      finalLinkText = this.props.i18n.go_to_code_studio;
-    }
-
-    return (
-      <table style={style}>
-        <tbody>
-          <tr>
-            <td>
-              <div style={circleStyle}>
-                <div style={exclamationStyle}>!</div>
-              </div>
-            </td>
-            <td style={bodyStyle}>
-              <AbuseError
-                i18n={this.props.i18n}
-                className="exclamation-abuse"
-                textStyle={textStyle}
-              />
-              <p className="exclamation-abuse" style={textStyle}>
-                <a href={finalLink}>{finalLinkText}</a>
-              </p>
-            </td>
-          </tr>
-        </tbody>
-      </table>
-    );
-  }
+  return (
+    <AlertExclamation>
+      <AbuseError i18n={i18n} className="exclamation-abuse" />
```

Comment:
It's interesting we're passing the whole i18n around, is this because different files use different i18n objects?

Reply:
It looks like [one string](https://github.com/code-dot-org/code-dot-org/blob/e0440b9891ca638c6c62c1d3014bbbe6d50df9a9/apps/src/code-studio/initApp/renderAbusive.js#L16) is different depending on the specific error so one string does need to be passed around. I don't know why the rest need to be though